libclamav/others_common.c

```diff
@@ -176,18 +177,31 @@ inline void cli_dbgmsg(const char *str, ...)
 {
     if (UNLIKELY(cli_get_debug_flag())) {
         MSGCODE(buff, len, "LibClamAV debug: ");
-        fputs(buff, stderr);
+        clrs_eprint(buff);
     }
 }
 
 void cli_dbgmsg_no_inline(const char *str, ...)
 {
     if (UNLIKELY(cli_get_debug_flag())) {
         MSGCODE(buff, len, "LibClamAV debug: ");
-        fputs(buff, stderr);
+        clrs_eprint(buff);
     }
 }
 
+size_t cli_eprintf(const char *str, ...)
+{
+    va_list args;
+    char buff[MSGBUFSIZ];
+    va_start(args, str);
+    vsnprintf(buff, sizeof(buff), str, args);
+    buff[sizeof(buff) - 1] = '\0';
+    va_end(args);
+    clrs_eprint(buff);
+
+    return strlen(buff);
```

Comment:
vsnprintf() returns the number of characters written.  Perhaps a light optimization based on that rather than scanning the string again?